nilearn/maskers/tests/test_nifti_labels_masker.py

```diff
@@ -7,68 +7,57 @@
 
 import warnings
 
-import nibabel
 import numpy as np
 import pytest
+from nibabel import Nifti1Image
+from numpy.testing import assert_almost_equal, assert_array_equal
 
-from nilearn._utils import data_gen, testing
+from nilearn._utils.data_gen import (
+    generate_labeled_regions,
+    generate_random_img,
+)
 from nilearn._utils.exceptions import DimensionError
+from nilearn._utils.testing import write_imgs_to_path
 from nilearn.image import get_data
 from nilearn.maskers import NiftiLabelsMasker, NiftiMasker
 
 
-def test_nifti_labels_masker():
+@pytest.fixture
+def n_regions():
+    return 9
+
+
+@pytest.fixture
+def length():
+    return 93
+
+
+def test_nifti_labels_masker(affine_eye, shape_3d_default, n_regions, length):
     """Check working of shape/affine checks."""
-    length = 3
-    shape1 = (13, 11, 12, length)
-    affine1 = np.eye(4)
+    shape1 = (*shape_3d_default, length)
 
     shape2 = (12, 10, 14, length)
-    affine2 = np.diag((1, 2, 3, 1))
-
-    n_regions = 9
 
-    fmri11_img, mask11_img = data_gen.generate_random_img(
-        shape1,
-        affine=affine1,
-    )
-    fmri12_img, mask12_img = data_gen.generate_random_img(
+    fmri11_img, mask11_img = generate_random_img(
         shape1,
-        affine=affine2,
+        affine=affine_eye,
```

Comment:
In the previous test, affine2 was distinct from affine_eye.

Reply:
yeah the diff is confusing on this one. 🙈 

the affine used to generate `fmri12_img` and `mask12_img` was `affine2`

the line now is to generate **`fmri11_img`** and **`mask11_img`** and this was `affine1=np.eye(4)`

Reply:
OK